packages/wonder-blocks-tooltip/components/tooltip-popper.js

```diff
@@ -37,14 +29,43 @@ type Props = {|
 |};
 
 export default class TooltipPopper extends React.Component<Props> {
-    _renderPositionedContent(childrenProps: PopperJSChildrenProps) {
-        const {children, placement} = this.props;
+    _bubbleRefTracker = new RefTracker();
+    _tailRefTracker = new RefTracker();
+
+    _getUpdatedBubbleRefTracker(targetRefFn: (?HTMLElement) => void) {
+        this._bubbleRefTracker.setCallback(targetRefFn);
+        return this._bubbleRefTracker.updateRef;
+    }
+
+    _getUpdatedTailRefTracker(targetRefFn: (?HTMLElement) => void) {
+        this._tailRefTracker.setCallback(targetRefFn);
+        return this._tailRefTracker.updateRef;
+    }
+
+    _renderPositionedContent(popperProps: PopperChildrenProps) {
+        const {children} = this.props;
 
         // We'll hide some complexity from the children here and ensure
         // that our placement always has a value.
-        childrenProps.placement = childrenProps.placement || placement;
+        const placement = popperProps.placement || this.props.placement;
+
+        // Just in case the callbacks have changed, let's update our reference
+        // trackers.
+        this._bubbleRefTracker.setCallback(popperProps.ref);
+        this._tailRefTracker.setCallback(popperProps.arrowProps.ref);
 
-        return children(childrenProps);
+        const bubbleProps = {
+            placement: placement,
+            style: popperProps.style,
+            updateBubbleRef: this._bubbleRefTracker.updateRef,
+            tailOffset: {
+                top: popperProps.arrowProps.style.top,
+                left: popperProps.arrowProps.style.left,
+            },
+            updateTailRef: this._tailRefTracker.updateRef,
+            outOfBoundaries: popperProps.outOfBoundaries,
+        };
+        return children(bubbleProps);
```

Comment:
I see... there is where the conversion from popper props to bubble props happens.  It might be worth calling this out with a comment so that it's easy to tell where the boundary between our code and react-popper code is.

Reply:
Yes, fair. I will do that.